### PR TITLE
Mathias/persist token awaiting

### PR DESCRIPTION
**Pull request by mathiasquintero**
### Issue
This fixes the following issue(s): 
https://github.com/TCA-Team/iOS/issues/123

The UI is a bit funky.
Maybe @TG908 or @rodalfus can help.

Just fyi the button looks weird when we have already requested a token and awaiting confirmation

### CODE REVIEW

**Comment by rodalfus** [comment]
@mathiasquintero I guess the issue is that we started using a shared instance of the login view controller flow with #139 (underlying issue) / #143 (PR). That means that the button gets animated to be “small” / gets reset to that smaller position and seems to be kept in memory. What we could do about this is: 1) stop keeping the login view controllers in memory 2) reset the button to its initial position after the animation is done.

File: TUM Campus App/PersistentUserData.swift [hunk 1]
```diff
@@ -10,20 +10,29 @@ import Sweeft
 
 enum PersistendUserData: Codable {
     
+    enum State: Int, Codable {
+        case loggedIn
+        case awaitingConfirmation
+    }
```
Comment (mammuth) [inline TUM Campus App/PersistentUserData.swift]: 👍

File: TUM Campus App/WaitForTokenViewController.swift [hunk 2]
```diff
@@ -42,30 +36,33 @@ extension WaitForTokenViewController {
...
+        } else {
+            button.startLoadingAnimation()
+            loginManager.fetch()
+        }
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        guard !PersistentUser.isLoggedIn else { return }
+        PersistentUser.reset()
```
Comment (mammuth) [inline TUM Campus App/WaitForTokenViewController.swift]: So when I login, get to the WaitForToken Screen and press the Home button (without authorising the token yet), the PersistentUser will get reset? Shouldn't the state still be .awaitingConfirmation ?

**Comment by mammuth** [comment]
You could also think about merging master in order to get the CI up and getting feedback by our bot (if you dare showing your code to him 🙈😆)

File: TUM Campus App/WaitForTokenViewController.swift (hunk 2 above)
Comment (mathiasquintero) [reply TUM Campus App/WaitForTokenViewController.swift]: Ah, I forgot. Does `viewWillDisapper` get called when pressing the home button?

Comment (mathiasquintero) [reply TUM Campus App/WaitForTokenViewController.swift]: I checked. I doesn't get called

Comment (mammuth) [reply TUM Campus App/WaitForTokenViewController.swift]: Alright! 😊

**Pull request merged**